### DISCUSSION TITLE
Javascript file is not loading in the github pages

### DISCUSSION THREAD

**Discussion by RSathya16** [community/community#68439] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have uploaded a simple HTML,CSS,JS website in github Pages. The CSS file ,Image file and HTML file is loading fine and it is showing in the sources tab in chrome

![image](https://github.com/community/community/assets/142917390/333e2160-7ad3-4f60-95bf-0ca9503ac732)

but my js file is not there it is only available in the Application tab

![image](https://github.com/community/community/assets/142917390/4c4784c8-c2ba-4813-b509-bb4e8aaa7c5c)

I have tried many ways but I can't find the solution Javascript part is not working still.

The hosted link : https://rsathya16.github.io/clothing-ecommerce/#

Anyone can help...!

Thanks in Advance.

**Comment by melvincwng** [comment]
Hey @RSathya16, from my understanding, GitHub pages are cached by default via a CDN.

From what I observed in the Chrome dev tools, your website - https://rsathya16.github.io/clothing-ecommerce/# is still showing the **old** index.html file.

Your **new** index.html file (along with your latest repository changes) seems to be working - I've tested it by spinning an instance of your repository on GitHub Codespace.

My suggestion would be to:
- First, try clearing your browser cache. 
- See if your website works. If it doesn't, probably it's still fetching the old index.html file.
- If the website still does not work after clearing your browser cache, **kindly wait for the cache (on GitHub's CDN side) to expire on its own**. 
- Once that cache expires, your website should be working as it should be retrieving the updated new index.html file

Hope this resolves your issue and answers your question.

Take care :smile:

**Reply by melvincwng** [reply]
Alternatively, try republishing your website by following the instructions in this link - [How to republish GitHub pages by IsmailUmmer](https://github.com/orgs/community/discussions/29619#:~:text=if%20u%20lost%20the%20republish%20button).

**Reply by RSathya16** [reply]
I have tried above methods which you have suggested but. still the script file is not loading

and you said new index file is not loaded in chrome dev tool but it is loading my new index file. it is fetching and loading all my latest changes which I have made in the index file or style.css file. the problem is JS file is still not loading.

this is my latest changes which I did yesterday (I have imported the script file inside the head tag before it was inside the body tag)

![image](https://github.com/community/community/assets/142917390/e4f6953f-570d-4ca8-b16b-7ab768932b61)

**Comment by RSathya16** [comment]
I found the cause of the issue .

I am trying to import the js file in the index.html by using the script tag which type is "**module**" . If I change the type to "text/javascript" the Js is loading properly.

but I need the script as module because I am importing other json and js file inside the mainscript.js file (**while we using the import inside js file  we need that  script tag type as module**).

how to resolve this any idea..?

**Comment by melvincwng** [comment, marked as answer]
Hi @RSathya16, after debugging your website for some time, I noticed the root cause of the issue.

As you can see in the screenshot below, it is unable to find these 2 files - hence the JS code does not work:

![image](https://github.com/community/community/assets/77479885/e5ad628c-7943-4422-a57d-9fa36ff8d6a3)

GitHub pages uses a static site generator called Jekyll to create your website. However, behind the scenes, folders that begin with a `.` will not be included in the final build. In your repository, those 2 files in the screenshot were in a folder called `.vscode`. Since that folder name starts with a `.`, it was not included in the final build by Jekyll, hence without those files, the JS code did not work.

Now that we know what caused the problem, how do we fix it? There are a couple of solutions you can try:

1. Adding a `.nojekyll` file in the root of your code repository.
> - This is probably the **easiest** solution.
> - It's essentially an empty file with an extension called `.nojekyll` 
> - The purpose of adding this file is to bypass the default Jekyll build process that GitHub uses, thus allowing folders preceding with the `.` symbol (i.e. .vscode) to be in the final build. 
> - Thus when those files are found, your JS code should run and work.
> - <img src="https://github.com/community/community/assets/77479885/22ae366d-c3a2-4fb7-bdf9-b88ab61462f0" alt="image" width="120px" height="40px" />

2. If solution 1 does not work or you are not familiar with Jekyll settings, consider this - rename the import paths of line 1 and line 2 in `mainscript.js`.
> - Instead of `import productAPI from "./.vscode/api.json" assert { type: "json" };` and `import closepopup from "./.vscode/closepoup.js";`
> - Change to `import productAPI from "./api.json" assert { type: "json" };` and `import closepopup from "./closepopup.js";`
> - **Take note**, there also seems to be a spelling error in one of your closepopup.js file (make sure to reference the correct file).
> - The JS file is called `closepopup.js` whereas the one found in the .vscode folder is called `closepoup.js` (subtle difference).
> - Changing the import paths should allow those files to be found since those files can also be found in the root of your directory thus theoretically speaking, it should result in your JS code to work.

3. If either Solution 1 or 2 fails, try this - rename the folder `.vscode` to something else like `vscode` (so that Jekyll can include it in the final build). Make sure to change the respective import paths accordingly as well in your `mainscript.js` file.

Hope either of these 3 solutions work for you and resolve your issue.

Take care 👍🏻 

Additional reference
- [How Jekyll works](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll)

**Reply by RSathya16** [reply]
Thank you so much @melvincwng I forgot to change the path 
"**import productAPI from "./.vscode/api.json" assert { type: "json" }; and import closepopup from "./.vscode/closepoup.js";**"

I know the file name prefixed with "." wont take in git hub pages that's why I have created the files outside that folder but forgot to change the path. now I have removed that **.vscode** from the path so it is working fine now 

and I will came to know about jekyll" by this discussion Thanks for that @melvincwng .

Have a nice day 👍🏻

**Reply by melvincwng** [reply]
@RSathya16 pleasure in helping 😀

**Comment by srijah599** [comment]
i made a new repository and while uploading the files the HTML file is uploaded fine but the js and CSS file is not uploading .what to do?

**Reply by Nitiien** [reply]
check for any spelling mistake or u just don't put them into a folder or run git commit .
